addons/l10n_cl/models/res_partner.py

```diff
@@ -8,17 +8,23 @@ class ResPartner(models.Model):
     _name = 'res.partner'
     _inherit = 'res.partner'
 
+    def _get_default_l10n_cl_sii_taxpayer_type(self):
+        allowed_company = self._context.get('allowed_company_ids')
+        if not allowed_company:
+            return self._context.get('install_module') == 'l10n_cl' and '1'
+        company = self.env['res.company'].browse(allowed_company[0])
+        return company.country_id == self.env.ref('base.cl') and '1'
```

Comment:
I think partners are created without company by default.  Even if the partner is not related to Chile, there is no problem in having a '1'.  So, is this change really needed?

Reply:
This is needed because in tests (Rubi sent us a video), in a multiloc environment you were creating a partner from a different country to CL, the taxpayer type was established to 1 (remember that the field is being shown in the partner form).

Reply:
@Danisan We said that it was no problem to have a '1' everywhere.  And in theory it is, but of course it can be weird for the user if he sees the field.  But this code is going too far.  I would rather hide the field if the country is Chile then (but that is a bit against what we do with all the rest, although if it is only needed for Chilean companies it can make sense)  @ren-odoo

Reply:
@jos-odoo if the criteria are to hide fields there are several fields in the same condition. To hide each one individually we have a lot of work to do. 

Nevertheless, there is a proposal from us with a filter by localization. Did you consider this?

Reply:
@jco-odoo Thinking in the multi company/mutlilocalization approach my only concern if we hide the field is the next scenario: 
If a contact is created in an Argentinean company and the Chilean taxpayer type is not displayed or filled with a default value then if at some point the same contact needs to be used on and Chilean company we can face blocking errors on the invoice because the document type depends on this field.  This is might be a corner case but let me know if you think it can affect this decision. 
cc. @Danisan

Reply:
Hi @Danisan after analyzing it with Jos we decide that the best approach is the next (this is gonna be the same for l10n_ar so we can have the same standard):
- It's ok to display the field regardless of the partner company. 
- The default value should be empty so the users can fill it only when necessary (it shouldn't represent an issue on the invoice  as we have the preventing message in the invoice when this field is empty for a partner)

Reply:
@jco-odoo @ren-odoo and @suv-bmya fixed in bab4b6b